File: leetcode_2530/MaximalScoreAfterApplyingKOperations.py
import heapq
import math

class MaximalScoreAfterApplyingKOperations:
    def max_k_elements(self, nums: list[int], k: int) -> int:
        res = 0
        max_heap = [-n for n in nums]
        heapq.heapify(max_heap)

        while k:
            n = -heapq.heappop(max_heap)
            res += n
            k -= 1
            heapq.heappush(max_heap, -math.ceil(n / 3))
            
        return res
    
def main():
    main_instance = MaximalScoreAfterApplyingKOperations()
    print(main_instance.max_k_elements(nums = [1,10,3,3,3], k = 3))

if __name__ == "__main__":
    main()
       

File: leetcode_2530/MaximalScoreAfterApplyingKOperations.java
package leetcode_2530;

import java.util.PriorityQueue;

class MaximalScoreAfterApplyingKOperations{
    public long maxKElements(int[] nums, int k) {
        long res = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b - a);
        for (int num : nums)
            pq.offer(num);
        
        while (!pq.isEmpty() && k > 0)
        {
            res += pq.peek();
            int top = (int)Math.ceil(pq.poll() / 3.0);
            pq.offer(top);
            k--;
        }
        return res;
    }

    public static void main(String[] args) {
        MaximalScoreAfterApplyingKOperations msako = new MaximalScoreAfterApplyingKOperations();
        int[] nums = {1,10,3,3,3};
        int k = 3;
        System.out.println(msako.maxKElements(nums, k));
    }
}